### PR TITLE
CASSANDRA-18683: Add PartitionSizeTableProvider for reading the compressed and uncompr…

### PR DESCRIPTION
**Pull request by jberragan**
…essed sizes of all partitions in a table by utilizing the SSTable Index.db files

### CODE REVIEW

File: cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/sparksql/SparkCellIterator.java [hunk 1]
```diff
@@ -339,21 +337,28 @@ private void maybeRebuildPartition()
...
+                                        final CqlTable table,
+                                        final Object[] values,
+                                        final Stats stats)
+    {
+        if (table.numPartitionKeys() == 1)
         {
             // Not a composite partition key
-            CqlField field = cqlTable.partitionKeys().get(0);
-            values[field.position()] = deserialize(field, partitionKey);
+            final CqlField field = table.partitionKeys().get(0);
```
Comment (yifan-c) [inline cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/sparksql/SparkCellIterator.java]: nit: avoid using the unnecessary `final`. The project use the same code style as in https://cassandra.apache.org/_/development/code_style.html

File: cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/sparksql/PartitionSizeIterator.java [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        if (it.hasNext())
+        {
+            it.advanceToNextColumn();
+
+            final IndexEntry entry = it.rid();
+            final Object[] values = new Object[numPartitionKeys + 2];
+
+            SparkCellIterator.readPartitionKey(entry.getPartitionKey(), cqlTable, values, stats);
+            values[numPartitionKeys] = entry.getUncompressed();
+            values[numPartitionKeys + 1] = entry.getCompressed();
```
Comment (yifan-c) [inline cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/sparksql/PartitionSizeIterator.java]: nit: add a comment to associate the data population with the schema defined at `partitionSizeStructType` in `DataLayer`.

**Review by yifan-c — COMMENTED** [review]
LGTM. A few nits.

File: cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/sparksql/SparkCellIterator.java (hunk 1 above)
Comment (jberragan) [reply cassandra-analytics-core/src/main/java/org/apache/cassandra/spark/sparksql/SparkCellIterator.java]: I don't see this check style in https://github.com/apache/cassandra-analytics/blob/trunk/config/checkstyle/checkstyle.xml

**Comment by jberragan** [comment]
Committed: https://github.com/apache/cassandra-analytics/commit/014db08a79f00ef0d94e6855779e398c9dc689c1

Jira: https://issues.apache.org/jira/browse/CASSANDRA-18683

**Pull request closed without merging**